packages/caravan-fees/src/rbf.ts

```diff
@@ -0,0 +1,303 @@
+import { PsbtV2 } from "@caravan/psbt";
+import BigNumber from "bignumber.js";
+import { DEFAULT_DUST_THRESHOLD, RBF_SEQUENCE } from "./constants";
+import { RbfOptions, FeeRate } from "./types";
+
+/**
+ * RbfTransaction class for handling Replace-By-Fee transactions
+ */
+class RbfTransaction {
+  private psbt: PsbtV2;
+  private newFeeRate: FeeRate;
+  private dustThreshold: string;
+
+  /**
+   * Create a new RbfTransaction
+   * @param psbt - The original transaction as a PsbtV2
+   * @param newFeeRate - The new fee rate for the replacement transaction
+   * @param options - Additional options for RBF
+   *   - dustThreshold: The minimum amount (in satoshis) for an output to not be considered dust.
+   *If not provided, a default value is used.
+   */
+  constructor(
+    psbt: PsbtV2,
+    newFeeRate: FeeRate,
+    options?: { dustThreshold?: string }
+  ) {
+    this.psbt = new PsbtV2(psbt.serialize("base64"));
```

Comment:
I would suggest just passing in a raw psbt keeping the api boundary as clean as possible. The way you have it now the consumer has to be using the psbt v2 package directly. If it's a raw psbt then you can convert that on class initialization. 

I would recommend adding some validation in the constructor that catches errors. You could also handle different psbt versions (PsbtV2 has a static method to convert from V0) which adds even more flexibility.

Reply:
Done [here](https://github.com/caravan-bitcoin/caravan/commit/77f0cbeb67a274b0aea5a69e87168b9ac1e2304a#diff-47122bb461c82a41c84d8866adf0cb005bf7f573a2afb8558f670a419bed146aR51)